### PR TITLE
Add configuration so you can develop in a VSCode dev container or Github Codespaces

### PR DESCRIPTION
**Pull request by talk2MeGooseman**
I hope you dont mind this addition. I wanted to make it even easier for someone to get this project up and running at start hacking away at AoC problem by simplifying this environment setup.

This change creates a build of an Elixir docker image with pre-installed packages for everyday development. Then using the `devcontainer.json` file, you can either have Visual Studio Code build and SSH the Docker image locally to start coding away or launch the project in Github Codespaces and work in the cloud. No need to install anything locally since its all inside the Docker image and development is done in the running container.

Let me know what you think.

### CODE REVIEW

**Comment by mhanberg** [comment]
This looks cool!

Could you add some documentation to the README regarding this? I don't use VSCode, so I am not sure how to verify that this works.

**Comment by talk2MeGooseman** [comment]
No problem! Ill get that done soon.

**Comment by talk2MeGooseman** [comment]
@mhanberg Update to README made, let me know if it all makes sense.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+ARG VARIANT=latest
+FROM elixir:${VARIANT}
```
Comment (mhanberg) [inline .devcontainer/Dockerfile:2]:
```suggestion
FROM elixir:latest
```
I installed vscode and tried to boot this up and it installed elixir 1.7. Hard coding this as latest fixed it for me.

Comment (talk2MeGooseman) [reply .devcontainer/Dockerfile:2]: Ah thanks for the heads up, I made a change to the config file to used `latest` also added a setup in the guide calling this out if people would like to set a specific version of Elixir

**Comment by mhanberg** [comment]
Thanks!

**Pull request merged**